draco/core/containers.py

```diff
@@ -366,6 +379,51 @@ def _make_selections(self, sel_args):
 
         return selections
 
+    def copy(self, shared=None):
+        """Copy this container, optionally sharing the source datasets.
+
+        This routine will create a copy of the container. By default this is
+        as full copy with the contents fully independent. However, a set of
+        dataset names can be given that will share the same data as the
+        source to save memory for large datasets. These will just view the
+        same memory, so any modification to either the original or the copy
+        will be visible to the other. This includes all write operations,
+        addition and removal of attributes, redistribution etc. This
+        functionality should be used with caution and clearly documented.
+
+        Parameters
+        ----------
+        shared : list, optional
+            A list of datasets whose content will be shared with the original.
+
+        Returns
+        -------
+        copy : subclass of ContainerBase
+            The copied container.
+        """
+        new_cont = self.__class__(attrs_from=self, axes_from=self, skip_datasets=True)
+
+        # Loop over datasets that exist in the source and either add a view of
+        # the source dataset, or perform a full copy
+        for name, data in self.datasets.items():
+
+            if shared and name in shared:
+                # TODO: find a way to do this that doesn't depend on the
```

Comment:
maybe you could use `MemDataset.view` on the right side, but I don't know how to access the left-hand side another way...

Reply:
That's what I had originally in here. However you don't want to do that because although initially it shares a copy of the data, as soon as you redistribute it doesn't. That's because it they only share the internal `._data` in the MemDataset which gets replaced during redistribution. It's subtle. Probably should comment on that explicitly.